tests/system_tests_delivery_counts.py

```diff
@@ -1001,24 +1007,35 @@ def _cleanup(self):
                 self.timer.cancel()
                 self.timer = None
 
+        def _set_receiver_stats(self, li):
+            self.receiver_stats = li
+            self.sender_stats = get_link_info("Tx_Test01", self.router_addr)
+            self._cleanup()
+
         def poll_timeout(self):
             """
             Periodically check the deliveryCount on the receiver.  Once it
             reaches rx_limit the test is complete: gather link statistics
             before closing the clients
             """
             li = get_link_info("Rx_Test01", self.router_addr)
-            if li and li['deliveryCount'] == self.rx_limit:
-                self.receiver_stats = li
-                self.sender_stats = get_link_info("Tx_Test01", self.router_addr)
-                self._cleanup()
+            if li and self.outcome == Delivery.RELEASED and li['releasedCount'] == self.rx_limit:
```

Comment:
Ah - yes this is better... but...
If presettled == false, then the last counter to be updated would be the sender's counters, since it takes time to propagate the outcome back to the sender the sender's outcomes would lag behind the receivers.

Perhaps it would be best to simply keep polling the statistics for both sender and receiver and exit the poll loop once the counters haven't changed since the last poll?